proxy/src/proxy_service.rs

```diff
@@ -86,32 +86,50 @@ where
 
         println!("Resolved route: {route:?}");
 
+        // Extract owned data from the route match before moving request
+        let route_info: Option<(config::Action, std::collections::HashMap<String, String>)> = route
+            .map(|RouteMatch { action, params }| {
+                let owned_params = params
+                    .into_iter()
+                    .map(|(k, v)| (k, v.to_string()))
+                    .collect();
+                (action.clone(), owned_params)
+            });
+
         let upstreams = self.upstreams.clone();
         let resolvers = self.resolvers.clone();
         let client = self.client.clone();
 
         Box::pin(async move {
-            let upstream_name: Option<String> = match route {
-                Some(RouteMatch { action, params }) => match action {
+            let upstream_name: Option<String> = match route_info {
+                Some((action, params)) => match action {
                     config::Action::Static { to } => Some(to),
                     config::Action::Dynamic {
                         resolver,
                         cell_to_upstream,
                         default,
                         ..
-                    } => resolvers
-                        .resolve(&resolver, &cell_to_upstream, params)
-                        .await
-                        .ok()
-                        .map(|s| s.to_string())
-                        .or(default),
+                    } => {
+                        // Convert params back to borrowed form for the resolver
+                        let borrowed_params: std::collections::HashMap<String, &str> = params
+                            .iter()
+                            .map(|(k, v)| (k.clone(), v.as_str()))
+                            .collect();
```

Comment:
It seems like `resolve` has changed in a way that requires these extra clones + boilerplate for users of that function.. can we avoid that?